src/entt/entity/registry.hpp

```diff
@@ -833,8 +843,12 @@ class basic_registry {
      */
     template<typename... Component, typename It>
     void remove(It first, It last) {
-        ENTT_ASSERT(std::all_of(first, last, [this](const auto entity) { return valid(entity); }));
-        (assure<Component>().remove(*this, first, last), ...);
+        if constexpr(sizeof...(Component) == 0) {
+            std::for_each(first, last, [this](const auto entity) { this->remove(entity); });
```

Comment:
Nitpick: I'd use a for loop here to reduce compile time.

Reply:
I see that e62f1edada84f1b029fe5fa1dba303d4ab3d76fb recently replaced std::for_each as I was pretty sure I didn't make that up… I "just" copied and rebased in the wrong order.
```suggestion
            while(first != last) { remove(*(first++)); }
```
No idea of the compile time effects & I see std::for_each used in a few more places (not in that file though). If this (and perhaps other things) are forbidden [function poisoning](https://www.fluentcpp.com/2018/09/04/function-poisoning-in-cpp/) might be a good idea to stop it from reappearing.